contracts/c-sphincs-all-in-one-lock/nist-vector-tester.c

```diff
@@ -0,0 +1,113 @@
+/*
+ * This is a lock only used for testing against NIST test vectors.
+ * Don't use it in production!
+ *
+ * It expects a variable length message in the first witness of the transaction
+ * (not script group), the second witness of transaction should contain param
+ * id, public key and signature.
+ *
+ * This format is purposely designed so we can pass each NIST test vector to a
+ * CKB transaction to verify.
+ *
+ * It delegates to one of the leaf locks via exec syscalls to
+ * do the actual sphincs+ signature verification.
+ */
+
+/*
+ * The current project is architected so multiple C sources files
+ * are compiled respectively. However, ckb-c-stdlib requires only
+ * one C source file to include entry.h file. This trick here ensures
+ * that only current source file will include entry.h, which also includes
+ * all the libc implementations.
+ */
+#undef CKB_DECLARATION_ONLY
+#include "entry.h"
+#define CKB_DECLARATION_ONLY
+
+#include "ckb-sphincsplus-common.h"
+
+#include <ckb_syscalls.h>
+#include <molecule/blockchain.h>
+
+#define CKB_SCRIPT_MERGE_TOOL_DEFINE_VARS
+#include "params-finder.h"
+
+#include "zero_escape_encoding.h"
+
+#define MAX_SCRIPT_SIZE (1024)
+#undef MAX_WITNESS_SIZE
+#define MAX_WITNESS_SIZE (512 * 1024)
+
+int fetch_params(uint8_t id, const CkbSphincsParams **params) {
+  int err = CKB_SUCCESS;
+  uint8_t index = MULTISIG_PARAM_ID_TO_INDEX(id);
+  CHECK2(index >= 0 && index < CKB_SPHINCS_SUPPORTED_PARAMS_COUNT,
```

Comment:
This is redundant checking, the type of index is `uint8_t`, so `index >= 0` always holds.

Reply:
Done